%\VignetteIndexEntry{soggi}
%\VignettePackage{soGGi}
%\VignetteEngine{knitr::knitr}

% To compile this document
% library('knitr'); rm(list=ls()); knit('soggi.Rnw')

\documentclass[12pt]{article}\usepackage[]{graphicx}\usepackage[usenames,dvipsnames]{color}
%% maxwidth is the original width if it is less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.345, 0.345, 0.345}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.686,0.059,0.569}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.192,0.494,0.8}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.345,0.345,0.345}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.161,0.373,0.58}{\textbf{#1}}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0.69,0.353,0.396}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.333,0.667,0.333}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.737,0.353,0.396}{\textbf{#1}}}%

\usepackage{framed}
\makeatletter
\newenvironment{kframe}{%
 \def\at@end@of@kframe{}%
 \ifinner\ifhmode%
  \def\at@end@of@kframe{\end{minipage}}%
  \begin{minipage}{\columnwidth}%
 \fi\fi%
 \def\FrameCommand##1{\hskip\@totalleftmargin \hskip-\fboxsep
 \colorbox{shadecolor}{##1}\hskip-\fboxsep
     % There is no \\@totalrightmargin, so:
     \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
 \MakeFramed {\advance\hsize-\width
   \@totalleftmargin\z@ \linewidth\hsize
   \@setminipage}}%
 {\par\unskip\endMakeFramed%
 \at@end@of@kframe}
\makeatother

\definecolor{shadecolor}{rgb}{.97, .97, .97}
\definecolor{messagecolor}{rgb}{0, 0, 0}
\definecolor{warningcolor}{rgb}{1, 0, 1}
\definecolor{errorcolor}{rgb}{1, 0, 0}
\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX

\usepackage{alltt}




\RequirePackage{/Library/Frameworks/R.framework/Versions/3.1/Resources/library/BiocStyle/sty/Bioconductor}

\AtBeginDocument{\bibliographystyle{/Library/Frameworks/R.framework/Versions/3.1/Resources/library/BiocStyle/sty/unsrturl}}





\author{Thomas Carroll$^{1*}$\\[1em] \small{$^{1}$ Bioinformatics Facility, MRC Clincal Sciences Centre;} \\ \small{\texttt{$^*$thomas.carroll (at)imperial.ac.uk}}}

\title{Visualisation, transformations and arithmetic operations for grouped genomic intervals}
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}







\maketitle

\begin{abstract}
 
The soGGi package provides tools to summarise sequence data, genomic signal and motif occurrence over grouped genomic intervals as well to perform complex subsetting, arithmetic operations and transformations on these genomic summaries prior to visualisation.


As with other Bioconductor packages such as CoverageView and seqPlots. soGGi plots average signal across groups of genomic regions. soGGI provides flexibity in both its data aquisition and visualisation. Single or paired-end BAM files, bigWigs, rleLists and PWM matrices can be provided  as input alongside a GRanges object or BED file location. soGGi can plot summarises across actual size and normalised features such as genomic intervals of differing length ( as with genes). The use of normalised size plots for genes can however obsure high resolution events around the TSS. To address this, combination plots can be created within soGGI allowing for fine detail at the edges of normalised regions.


Arithmetic operation and transformations can be easily performed on soGGi objects allowing for rapid operations between profiles such as the substraction of input signal or quantile normalisation of replicates within a group. 


soGGi integrates the ggplot2 package and add functionality to rapidly subset, facet and colour profiles by their overlaps with GenomicRanges objects or grouping by metadata column IDs. 


The plotting, arithmetic and tranformation functions within soGGi allow for rapid evaluation of groups of genomic intervals and provide a toolset for user-defined analysis of these summaries over groups.

  \vspace{1em}
  
  \end{abstract}



\newpage

\tableofcontents

\section{Standard workflow}

\subsection{The soggi function}

The \Rfunction{regionPlot()} function is used to summarise signal, reads or PWM occurrence over grouped genomic intervals. Input can be BAM, bigWig or a PWM matrix and the regions to summarise over a character string of path to BED file or a GRanges object. The full set of soggi function arguments can be found in help pages.


In this example, signal coverage is summarised from a BAM file using the defaults. The default style of region plot is to produce a normalised by size region plot.

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{library}\hlstd{(soGGi)}
\hlstd{chipExample} \hlkwb{<-} \hlkwd{regionPlot}\hlstd{(}\hlstr{"pathToBAM/mybam.bam"}\hlstd{,myGRangesObject,}\hlkwc{format}\hlstd{=}\hlstr{"bam"}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}

A pre-computed data set is included in the package containing averages profiles created with command above for DNAse, Pol2, H3k9ac and H3k3me3. The object itself cantains all counts along interval region windows in assays slots and information of the samples in exptData slot accessible by \Rfunction{assays()} and \Rfunction{exptData} respectively.


\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{library}\hlstd{(soGGi)}
\hlkwd{data}\hlstd{(chipExampleBig)}
\hlstd{chipExampleBig}
\end{alltt}
\begin{verbatim}
## class: ChIPprofile 
## dim: 201 300 
## exptData(2): names AlignedReadsInBam
## assays(10): '' '' ... '' ''
## rownames: NULL
## rowData metadata column names(5): name biotype Feature giID giID.1
## colnames(300): Start-1 Start-2 ... End+99 End+100
## colData names(0):
\end{verbatim}
\end{kframe}
\end{knitrout}

This object contains 10 sets of profiles for 200 genes. The object can be subset using [[ to select samples of interest.

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{chipExampleBig[[}\hlnum{1}\hlstd{]]}
\end{alltt}


{\ttfamily\noindent\color{warningcolor}{\#\# Warning: closing unused connection 3 (vignettes/soggi.Rnw)}}\begin{verbatim}
## class: ChIPprofile 
## dim: 201 300 
## exptData(2): names AlignedReadsInBam
## assays(1): ''
## rownames: NULL
## rowData metadata column names(5): name biotype Feature giID giID.1
## colnames(300): Start-1 Start-2 ... End+99 End+100
## colData names(0):
\end{verbatim}
\begin{alltt}
\hlstd{chipExampleBig}\hlopt{$}\hlstd{highdnase}
\end{alltt}
\begin{verbatim}
## class: ChIPprofile 
## dim: 201 300 
## exptData(2): names AlignedReadsInBam
## assays(1): ''
## rownames: NULL
## rowData metadata column names(5): name biotype Feature giID giID.1
## colnames(300): Start-1 Start-2 ... End+99 End+100
## colData names(0):
\end{verbatim}
\end{kframe}
\end{knitrout}

Similarly profile objects can be concatenated or bound together using c and rbind.


\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{c}\hlstd{(chipExampleBig[[}\hlnum{1}\hlstd{]],chipExampleBig[[}\hlnum{2}\hlstd{]])}
\end{alltt}
\begin{verbatim}
## class: ChIPprofile 
## dim: 201 300 
## exptData(2): names AlignedReadsInBam
## assays(2): '' ''
## rownames: NULL
## rowData metadata column names(5): name biotype Feature giID giID.1
## colnames(300): Start-1 Start-2 ... End+99 End+100
## colData names(0):
\end{verbatim}
\begin{alltt}
\hlkwd{rbind}\hlstd{(chipExampleBig[[}\hlnum{1}\hlstd{]],chipExampleBig[[}\hlnum{2}\hlstd{]])}
\end{alltt}
\begin{verbatim}
## class: ChIPprofile 
## dim: 402 300 
## exptData(2): names AlignedReadsInBam
## assays(1): ''
## rownames: NULL
## rowData metadata column names(5): name biotype Feature giID giID.1
## colnames(300): Start-1 Start-2 ... End+99 End+100
## colData names(0):
\end{verbatim}
\end{kframe}
\end{knitrout}


\subsection{Plotting profiles}

The \Rfunction{plotRegion()} function is used to produce profile plots. \Rfunction{plotRegion()} uses ggplot2 to generate plots and so returned object can be highly customisable using ggplot2 methods. 

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{plotRegion}\hlstd{(chipExampleBig[[}\hlnum{3}\hlstd{]])}
\end{alltt}
\end{kframe}
\end{knitrout}

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quicka-1.png}
\caption{
  \textbf{Example profile plot.}
  The plot generated by plotRegion() function on a single sample soGGi object. The x-axis shows the normalised length and the y-axis shows the coverage in windows. A Clear peak around the TSS can be observed for this Pol2 ChIP-seq profile.
}
\label{Example-1}
\end{figure}


When dealing with objects with multiple samples, the arguments groupBy and colourBy specify whether to facet or colour by Sample/Group respectively.

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{library}\hlstd{(ggplot2)}
\hlkwd{plotRegion}\hlstd{(chipExampleBig,}\hlkwc{colourBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,} \hlkwc{groupBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,} \hlkwc{freeScale}\hlstd{=}\hlnum{TRUE}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}


\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quick2-1.png}
\caption{
  \textbf{Multi-Sample profile plot.} 
  Here multiple samples are plotted simultaneously. Enrichment around TSS can be seen for all profiles with H3k9ac showing more enrichment in the gene body.
}
\label{multiplot}
\end{figure}



Here some samples can be seen to be noisy. Windsorisation can be applied when plotting using the outliers argument.

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{library}\hlstd{(ggplot2)}
\hlkwd{plotRegion}\hlstd{(chipExampleBig,}\hlkwc{colourBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,} \hlkwc{outliers}\hlstd{=}\hlnum{0.01}\hlstd{,} \hlkwc{groupBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,}\hlkwc{freeScale}\hlstd{=}\hlnum{TRUE}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quick2b-1.png}
\caption{
  \textbf{Multi-Sample profile plot with windsorisation.} 
  This multi-sample plot has windsorisation applied to outliers. The resulting plot is smoother than that seen in figure 2.
}
\label{mutliplot_Win}
\end{figure}

The \Rfunction{plotRegion()} can also be used to group genomic intervals while plotting using the \Rfunction{gts} argument. The \Rfunction{gts} argument either takes a GRanges object or a list of character vectors and the \Rfunction{summariseBy} argument to specify metadata to use.

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{library}\hlstd{(GenomicRanges)}
\hlstd{subsetsCharacter} \hlkwb{<-} \hlkwd{list}\hlstd{(}\hlkwc{first25} \hlstd{= (}\hlkwd{as.vector}\hlstd{(}\hlkwd{rowData}\hlstd{(chipExampleBig[[}\hlnum{1}\hlstd{]])}\hlopt{$}\hlstd{name[}\hlnum{1}\hlopt{:}\hlnum{25}\hlstd{])),} \hlkwc{fourth25} \hlstd{=} \hlkwd{as.vector}\hlstd{(}\hlkwd{rowData}\hlstd{(chipExampleBig[[}\hlnum{1}\hlstd{]])}\hlopt{$}\hlstd{name[}\hlnum{76}\hlopt{:}\hlnum{100}\hlstd{]))}

\hlstd{subsetsGRanges} \hlkwb{<-} \hlkwd{GRangesList}\hlstd{(}\hlkwc{low}\hlstd{=(}\hlkwd{rowData}\hlstd{(chipExampleBig[[}\hlnum{1}\hlstd{]])[}\hlnum{1}\hlopt{:}\hlnum{25}\hlstd{]),} \hlkwc{high}\hlstd{=}\hlkwd{rowData}\hlstd{(chipExampleBig[[}\hlnum{2}\hlstd{]])[}\hlnum{76}\hlopt{:}\hlnum{100}\hlstd{])}

\hlkwd{plotRegion}\hlstd{(chipExampleBig[[}\hlnum{1}\hlstd{]],}\hlkwc{gts}\hlstd{=subsetsCharacter,}\hlkwc{summariseBy} \hlstd{=} \hlstr{"name"}\hlstd{)}
\hlkwd{plotRegion}\hlstd{(chipExampleBig[[}\hlnum{1}\hlstd{]],}\hlkwc{gts}\hlstd{=subsetsGRanges)}
\end{alltt}
\end{kframe}
\end{knitrout}


\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quick3-1.png}
\caption{
  \textbf{DNAse grouped genomic intervals plot.} 
  This plots shows the plot for DNAse signal across normalised regions with separate profiles for each group of genomic intervals defined in gts.
.}
\label{gtsarguments}
\end{figure}

\newpage

\section{Transformations and arithmetic operations}

\subsection{Simple arithmetic operations on grouped profiles}


Common arithmetic operations and tranformations can be used with soGGi profile objects allowing for further  analysis post summarisation and iteratively over visualisations.

Here we summarise RNApol2 high and low and compare between replicates.


\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{pol_Profiles} \hlkwb{<-} \hlkwd{c}\hlstd{((chipExampleBig}\hlopt{$}\hlstd{highPol}\hlopt{+}\hlstd{chipExampleBig}\hlopt{$}\hlstd{midPol)}
\hlstd{, (chipExampleBig}\hlopt{$}\hlstd{highPol_Rep2}\hlopt{+}\hlstd{chipExampleBig}\hlopt{$}\hlstd{midPol_Rep2))}
\hlkwd{plotRegion}\hlstd{(pol_Profiles,}\hlkwc{colourBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,}\hlkwc{outliers}\hlstd{=}\hlnum{0.01}\hlstd{,} \hlkwc{groupBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,} \hlkwc{freeScale}\hlstd{=}\hlnum{TRUE}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quick4-1.png}
\caption{
  \textbf{Plotting arithmetic results.} 
  This plot demonstrates the ability to plot the profiles generated by the results of arithmetic operations. Here data is combined within replicate number and results plotted.
}
\label{arithmeticplotting}
\end{figure}

Common normalisations, log transformations and other mathematical functions such as mean() are also implemented to allow for the comparison within and between profiles.

In this example the profiles are log2 transformed with zeros being assigned the minimum value for that region.


\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{log2Profiles} \hlkwb{<-} \hlkwd{log2}\hlstd{(chipExampleBig)}

\hlkwd{plotRegion}\hlstd{(log2Profiles,}\hlkwc{colourBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,}\hlkwc{outliers}\hlstd{=}\hlnum{0.01}\hlstd{,} \hlkwc{groupBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,}\hlkwc{freeScale}\hlstd{=}\hlnum{TRUE}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quick6-1.png}
\caption{
  \textbf{Plotting log2 transformed profiles.} 
  This plots shows the resulting profiles after log2 transformation of ChIP-data across antibodies.
}
\label{log2plotting}
\end{figure}

From this log2 transformed data we can look at the difference between Pol2 profiles

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{log2Polhigh} \hlkwb{<-} \hlkwd{mean}\hlstd{(log2Profiles}\hlopt{$}\hlstd{highPol, log2Profiles}\hlopt{$}\hlstd{highPol_Rep2)}
\hlstd{log2Polmid} \hlkwb{<-} \hlkwd{mean}\hlstd{(log2Profiles}\hlopt{$}\hlstd{midPol, log2Profiles}\hlopt{$}\hlstd{midPol_Rep2)}
\hlstd{diffPol} \hlkwb{<-} \hlstd{log2Polhigh}\hlopt{-}\hlstd{log2Polmid}


\hlstd{diffh3k9ac} \hlkwb{<-} \hlstd{log2Profiles}\hlopt{$}\hlstd{highk9ac}\hlopt{-}\hlstd{log2Profiles}\hlopt{$}\hlstd{midk9ac}


\hlkwd{plotRegion}\hlstd{(}\hlkwd{c}\hlstd{(diffPol,diffh3k9ac),}\hlkwc{colourBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,}\hlkwc{outliers}\hlstd{=}\hlnum{0.01}\hlstd{,} \hlkwc{groupBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,} \hlkwc{freeScale}\hlstd{=}\hlnum{TRUE}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}


\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quick6b-1.png}
\caption{
  \textbf{Plotting differentials.} In this plot the log2 difference between high and low samples for H3k9ac and Pol2 replicates is shown.
}
\label{diffplot}
\end{figure}

Quantile normalisation of allow windows in regions between samples can allow for better better visual comparison of changes between conditions when dealing with larger numbers of replicates. Here for demonstration we apply it two samples but with real data higher sample numbers would be recommended.

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{normHighPol} \hlkwb{<-} \hlkwd{normalise}\hlstd{(}\hlkwd{c}\hlstd{(chipExampleBig}\hlopt{$}\hlstd{highPol, chipExampleBig}\hlopt{$}\hlstd{highPol_Rep2),} \hlkwc{method}\hlstd{=}\hlstr{"quantile"}\hlstd{,}\hlkwc{normFactors} \hlstd{=} \hlnum{1}\hlstd{)}
\hlstd{normMidPol} \hlkwb{<-} \hlkwd{normalise}\hlstd{(}\hlkwd{c}\hlstd{(chipExampleBig}\hlopt{$}\hlstd{midPol, chipExampleBig}\hlopt{$}\hlstd{midPol_Rep2),} \hlkwc{method}\hlstd{=}\hlstr{"quantile"}\hlstd{,}\hlkwc{normFactors} \hlstd{=} \hlnum{1}\hlstd{)}

\hlstd{normPol} \hlkwb{<-}\hlkwd{c}\hlstd{(normHighPol}\hlopt{$}\hlstd{highPol, normHighPol}\hlopt{$}\hlstd{highPol_Rep2, normMidPol}\hlopt{$}\hlstd{midPol, normMidPol}\hlopt{$}\hlstd{midPol_Rep2)}
\hlkwd{plotRegion}\hlstd{(normPol,}\hlkwc{colourBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,}\hlkwc{outliers}\hlstd{=}\hlnum{0.01}\hlstd{,} \hlkwc{groupBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,} \hlkwc{freeScale}\hlstd{=}\hlnum{TRUE}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}

\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quick7-1.png}
\caption{
  \textbf{Quantile normalisation.} In this toy example, data has been quantiled normalised within groups and the results plotted. This demonstrates the uniformity in data following quantile normalisation.
}
\label{quantilenormplot}
\end{figure}


%--------------------------------------------------
\section{Creating GRanges combinations for plotting}
%--------------------------------------------------



A common operation in the analysis of summaries over genomic intervals is to compare between different sets of grouped genomic intervals. soGGi includes helper functions to deal with grouped genomic intervals.

The \Rfunction{groupByOverlaps()} function creates all combinations of grouped genomic intervals from GRangesLists and so is useful to evaluate how summaries change over subclasses of genomic intervals (such as over co-occuring peak sets)

The \Rfunction{findconsensusRegions()} and \Rfunction{summitPipeline()} functions identifies consensus regions between GRanges objects and re-summits consensus region sets respectively. This approach has been previously implemented to identify reproducible peak sets between biological replciates.

\subsection{Grouping genomic interval sets and plotting results}


In this example, two antibodies for the transcription factor Ikaros are used to plot the signal over common and unique peaks for each antibody. First the peaks sets are defined using \Rfunction{groupByOverlaps()}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{data}\hlstd{(ik_Example)}
\hlstd{ik_Example}
\end{alltt}
\begin{verbatim}
## $ha
## GRanges with 200 ranges and 2 metadata columns:
##         seqnames               ranges strand   |            ID    Score
##            <Rle>            <IRanges>  <Rle>   |      <factor> <factor>
##     [1]     chr1   [5218713, 5219618]      *   |   MACS_peak_4    60.84
##     [2]     chr1   [6372427, 6373665]      *   |   MACS_peak_7   438.68
##     [3]     chr1   [6456792, 6457633]      *   |   MACS_peak_9    96.87
##     [4]     chr1   [7080191, 7080946]      *   |  MACS_peak_15    57.82
##     [5]     chr1   [9225736, 9227873]      *   |  MACS_peak_17   166.57
##     ...      ...                  ...    ... ...           ...      ...
##   [196]     chr1 [36765816, 36767310]      *   | MACS_peak_130    127.4
##   [197]     chr1 [36822798, 36824995]      *   | MACS_peak_131  1387.29
##   [198]     chr1 [36995290, 36996196]      *   | MACS_peak_132    73.84
##   [199]     chr1 [37044729, 37045925]      *   | MACS_peak_135   193.63
##   [200]     chr1 [37356201, 37357028]      *   | MACS_peak_138    66.39
##   ---
##   seqlengths:
##     chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
##       NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
## 
## $endo
## GRanges with 200 ranges and 2 metadata columns:
##         seqnames               ranges strand   |            ID    Score
##            <Rle>            <IRanges>  <Rle>   |      <factor> <factor>
##     [1]     chr1   [4695718, 4695917]      *   |   MACS_peak_2    98.76
##     [2]     chr1   [4775045, 4775288]      *   |   MACS_peak_3   100.91
##     [3]     chr1   [4775403, 4775658]      *   |   MACS_peak_4     50.4
##     [4]     chr1   [4775664, 4776101]      *   |   MACS_peak_5    150.6
##     [5]     chr1   [4797674, 4798084]      *   |   MACS_peak_6   651.89
##     ...      ...                  ...    ... ...           ...      ...
##   [196]     chr1 [24740900, 24741101]      *   | MACS_peak_211   301.48
##   [197]     chr1 [30929613, 30929871]      *   | MACS_peak_220   106.95
##   [198]     chr1 [30930510, 30931164]      *   | MACS_peak_221  1604.65
##   [199]     chr1 [31005552, 31005763]      *   | MACS_peak_222   174.71
##   [200]     chr1 [31006685, 31007192]      *   | MACS_peak_223   529.42
##   ---
##   seqlengths:
##     chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
##       NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
\end{verbatim}
\begin{alltt}
\hlstd{peakSetCombinations} \hlkwb{<-} \hlkwd{groupByOverlaps}\hlstd{(ik_Example)}
\hlstd{peakSetCombinations}
\end{alltt}
\begin{verbatim}
## $endo
## GRanges with 100 ranges and 1 metadata column:
##         seqnames               ranges strand   | grangesGroups
##            <Rle>            <IRanges>  <Rle>   |      <factor>
##     [1]     chr1   [4695718, 4695917]      *   |          endo
##     [2]     chr1   [4775045, 4775288]      *   |          endo
##     [3]     chr1   [4775403, 4775658]      *   |          endo
##     [4]     chr1   [4775664, 4776101]      *   |          endo
##     [5]     chr1   [4797674, 4798084]      *   |          endo
##     ...      ...                  ...    ... ...           ...
##    [96]     chr1 [21534723, 21534898]      *   |          endo
##    [97]     chr1 [21635845, 21636030]      *   |          endo
##    [98]     chr1 [21705049, 21705198]      *   |          endo
##    [99]     chr1 [21739213, 21739804]      *   |          endo
##   [100]     chr1 [21757663, 21757886]      *   |          endo
##   ---
##   seqlengths:
##     chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
##       NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
## 
## $ha
## GRanges with 137 ranges and 1 metadata column:
##         seqnames               ranges strand   | grangesGroups
##            <Rle>            <IRanges>  <Rle>   |      <factor>
##     [1]     chr1   [5218713, 5219618]      *   |            ha
##     [2]     chr1   [6372427, 6373665]      *   |            ha
##     [3]     chr1   [6456792, 6457633]      *   |            ha
##     [4]     chr1   [7080191, 7080946]      *   |            ha
##     [5]     chr1   [9225736, 9227873]      *   |            ha
##     ...      ...                  ...    ... ...           ...
##   [133]     chr1 [89527991, 89530046]      *   |            ha
##   [134]     chr1 [89768074, 89769026]      *   |            ha
##   [135]     chr1 [89837445, 89838176]      *   |            ha
##   [136]     chr1 [90598333, 90599360]      *   |            ha
##   [137]     chr1 [91540499, 91541383]      *   |            ha
##   ---
##   seqlengths:
##     chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
##       NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
## 
## $`ha-endo`
## GRanges with 63 ranges and 1 metadata column:
##        seqnames               ranges strand   | grangesGroups
##           <Rle>            <IRanges>  <Rle>   |      <factor>
##    [1]     chr1   [4847117, 4848878]      *   |       ha-endo
##    [2]     chr1   [5072908, 5073864]      *   |       ha-endo
##    [3]     chr1   [6204023, 6205748]      *   |       ha-endo
##    [4]     chr1   [6252357, 6253568]      *   |       ha-endo
##    [5]     chr1   [6395852, 6396924]      *   |       ha-endo
##    ...      ...                  ...    ... ...           ...
##   [59]     chr1 [24012021, 24013019]      *   |       ha-endo
##   [60]     chr1 [24684410, 24686139]      *   |       ha-endo
##   [61]     chr1 [24740575, 24741381]      *   |       ha-endo
##   [62]     chr1 [30929409, 30931590]      *   |       ha-endo
##   [63]     chr1 [31005112, 31007518]      *   |       ha-endo
##   ---
##   seqlengths:
##     chr1 chr10 chr11 chr12 chr13 chr14 ...  chr6  chr7  chr8  chr9  chrX  chrY
##       NA    NA    NA    NA    NA    NA ...    NA    NA    NA    NA    NA    NA
\end{verbatim}
\end{kframe}
\end{knitrout}


The output from \Rfunction{groupByOverlaps()} can then be used to subset precomputed profiles of HA and Endogenous ChIP signal. Here we apply a log2 transformation to the data before plotting and cleaning up profile with windsorisation


\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{data}\hlstd{(ik_Profiles)}
\hlstd{ik_Profiles}
\end{alltt}
\begin{verbatim}
## class: ChIPprofile 
## dim: 4800 401 
## exptData(2): names AlignedReadsInBam
## assays(2): '' ''
## rownames: NULL
## rowData metadata column names(3): ID Score giID
## colnames(401): Point_Centre-200 Point_Centre-199 ... Point_Centre199
##   Point_Centre200
## colData names(0):
\end{verbatim}
\begin{alltt}
\hlstd{log2Ik_Profiles} \hlkwb{<-} \hlkwd{log2}\hlstd{(ik_Profiles)}
\hlkwd{plotRegion}\hlstd{(log2Ik_Profiles,}\hlkwc{outliers}\hlstd{=}\hlnum{0.01}\hlstd{,}\hlkwc{gts}\hlstd{=peakSetCombinations,} \hlkwc{groupBy}\hlstd{=}\hlstr{"Group"}\hlstd{,}\hlkwc{colourBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,} \hlkwc{freeScale}\hlstd{=}\hlnum{TRUE}\hlstd{)}
\hlkwd{plotRegion}\hlstd{(log2Ik_Profiles[[}\hlnum{1}\hlstd{]]} \hlopt{-} \hlstd{log2Ik_Profiles[[}\hlnum{2}\hlstd{]] ,}\hlkwc{outliers}\hlstd{=}\hlnum{0.01}\hlstd{,}\hlkwc{gts}\hlstd{=peakSetCombinations,} \hlkwc{groupBy}\hlstd{=}\hlstr{"Group"}\hlstd{,} \hlkwc{colourBy}\hlstd{=}\hlstr{"Sample"}\hlstd{,} \hlkwc{freeScale}\hlstd{=}\hlnum{TRUE}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}


\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quick10-1.png}
\caption{
  \textbf{Signal over common and unique Ikaros peaks.} This plot shows that, as expected, common and unique peaks show different profiles for the Ikaros antibodies.
}
\label{IkExample}
\end{figure}


This confirms that common and unique peaksets have different levels of the separate antibody signals. This can be better demonstrated by subtracting to signal sets from each other and re-plotting over groups as seen in the final example.


\begin{figure}
\centering
\includegraphics[width=1\textwidth]{figure/quick10-2.png}
\caption{
  \textbf{Differential Ikaros profiles over common and unique sets.} 
  The plot of difference in log2 HA and Endogenous Ikaros signal over peaks shows the expected difference in Ikaros antibody signaland uniformity of signal of common peaks.
}
\label{DiffIkaros}
\end{figure}



\bibliography{library}

\end{document}
